addons/account/models/account_move.py

```diff
@@ -4402,6 +4402,10 @@ def _add_cash_basis_lines_to_exchange_difference_vals(lines, exchange_diff_move_
             :param lines:                   The account.move.lines to which fix the residual amounts.
             :param exchange_diff_move_vals: The current vals of the exchange difference journal entry.
             '''
+            if self.env['ir.config_parameter'].get_param('account.skip_fx_on_caba_base_lines', False):
+                # /!\ Note: Some l10n_** might no need these extra lines as it causes to unbalance the taxes.
+                # Taxes are already balanced in the method `_create_tax_cash_basis_moves` in account_reconcile_partial
+                return
```

Comment:
Isn't it something that depends on the very least on the company, and not on the database?

How does it work if you have one company that needs the FX and one that doesn't?

Reply:
@william-andre 

Are you suggesting adding a new field in the stable branch in order to fix this issue?

My proposal yet seeming liberal is still conservative as no new field is being added.

could you provide a proposal that fits the question you are arising?

The one that I could envision could be quite cumbersome and not that clear to be deployed.

Regards.

Reply:
I don't understand why you need to skip this method.

Suppose rates 1 MXN = 3 USD in 2016 and 1 MXN = 2 USD in 2017.

Suppose an invoice of 1200.0 USD with 20% tax in 2016 to get the following entries:
| Account  | USD |  MXN |
| ------------- | ------------- | ----|
| Product Sales  | -1200  | -400 |
| Transition Tax Account | -240  | -80 |
| Receivable | 1440  | 480 |

Fully reconciled the invoice using a payment in 2017 of 1440 (1200 + 240) USD. A cash basis entry is generated:
| Account  | USD |  MXN |
| ------------- | ------------- | ----|
| Tax Account  | -240  | -120 |
| Transition Tax Account | 240  | 120 |

CASE 1 : the `Transition Tax Account` is a reconcile one, both lines are reconciled together and an exchange difference is generated:
| Account  | USD |  MXN |
| ------------- | ------------- | ----|
| Exchange Account | 0  | -40 |
| Transition Tax Account | 0  | 40 |

If you take a look on the general ledger, the total of `Transition Tax Account` is 120.0.

CASE 2: `Transition Tax Account` is not a reconcile account so we need `_add_cash_basis_lines_to_exchange_difference_vals` to get the same result.

So imo, your taxes are NOT balanced at this point depending your configuration otherwise such extra lines will not be generated.

Reply:
A config param containing the id of the company in the key could already be a solution for instance.
Or the value of the param could be a comma separated list of ids where this applies. Or...

Reply:
@smetl 

![Screen Shot 2021-06-25 at 4 36 31](https://user-images.githubusercontent.com/7598010/123404950-fa74a780-d56e-11eb-8d44-dff3ba72fe0f.png)

Reply:
> A config param containing the id of the company in the key could already be a solution for instance.
> Or the value of the param could be a comma separated list of ids where this applies. Or...

@william-andre I like your idea. It quite clever, mine was quite cumbersome. I was envisioning concatenation and python string formating that would have rendered something complicated to deploy.

Reply:
@hbto Dude... Copy-pasting twice the same screenshot is not explaining why you are considering it's a custom behavior managed by localizations nor why you have multiple times the same exchange difference journal items on your database. 

I tested on runbot 14.0 and it works as expected.